lib/bamboo/adapters/sendgrid_adapter.ex

```diff
@@ -1,167 +1,12 @@
 defmodule Bamboo.SendgridAdapter do
-  @moduledoc """
-  Sends email using SendGrid's JSON API.
-
-  Use this adapter to send emails through SendGrid's API. Requires that an API
-  key is set in the config.
-
-  ## Example config
-
-      # In config/config.exs, or config.prod.exs, etc.
-      config :my_app, MyApp.Mailer,
-        adapter: Bamboo.SendgridAdapter,
-        api_key: "my_api_key"
-
-      # Define a Mailer. Maybe in lib/my_app/mailer.ex
-      defmodule MyApp.Mailer do
-        use Bamboo.Mailer, otp_app: :my_app
-      end
-  """
-
-  @default_base_uri "https://api.sendgrid.com/api"
-  @send_message_path "/mail.send.json"
-  @behaviour Bamboo.Adapter
-
-  alias Bamboo.Email
-
-  defmodule ApiError do
-    defexception [:message]
-
-    def exception(%{params: params, response: response}) do
-      filtered_params = params |> Plug.Conn.Query.decode |> Map.put("key", "[FILTERED]")
-
-      message = """
-      There was a problem sending the email through the SendGrid API.
-
-      Here is the response:
-
-      #{inspect response, limit: :infinity}
-
-      Here are the params we sent:
-
-      #{inspect filtered_params, limit: :infinity}
-
-      If you are deploying to Heroku and using ENV variables to handle your API key,
-      you will need to explicitly export the variables so they are available at compile time.
-      Add the following configuration to your elixir_buildpack.config:
-
-      config_vars_to_export=(
-        DATABASE_URL
-        SENDGRID_API_KEY
-      )
-      """
-      %ApiError{message: message}
-    end
-  end
+  @moduledoc false
 
   def deliver(email, config) do
```

Comment:
Since these are unused, they should be renamed to `_email` and `_config` so that you don't get build warnings.